homeassistant/components/media_player/philips_js.py

```diff
@@ -0,0 +1,206 @@
+"""
+Media Player component to integrate TVs exposing the Joint Space API.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/media_player.philips_js/
+"""
+import logging
+from datetime import timedelta
+import json
+
+import requests
+import voluptuous as vol
+
+from homeassistant.components.media_player import (
+    PLATFORM_SCHEMA, SUPPORT_SELECT_SOURCE, SUPPORT_TURN_OFF,
+    SUPPORT_VOLUME_STEP, SUPPORT_VOLUME_MUTE, MediaPlayerDevice)
+from homeassistant.const import (
+    STATE_ON, STATE_OFF, STATE_UNKNOWN, CONF_HOST, CONF_NAME)
+from homeassistant.util import Throttle
+import homeassistant.helpers.config_validation as cv
+
+_LOGGER = logging.getLogger(__name__)
+
+MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=30)
+
+SUPPORT_PHILIPS_JS = SUPPORT_TURN_OFF | SUPPORT_VOLUME_STEP | \
+                     SUPPORT_VOLUME_MUTE | SUPPORT_SELECT_SOURCE
+
+DEFAULT_DEVICE = 'default'
+DEFAULT_HOST = '127.0.0.1'
+DEFAULT_NAME = 'Philips TV'
+DEVICE_BASE_URL = 'http://{0}:1925/1/{1}'
+DEVICE_NAME_URL = 'http://{0}:1925/1/system/name'
+DEVICE_INPUT_URL = 'http://{0}:1925/1/input/key'
+DEVICE_SRC_SET_URL = 'http://{0}:1925/1/sources/current'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_HOST, default=DEFAULT_HOST): cv.string,
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+})
+
+
+# pylint: disable=unused-argument
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the Philips TV platform."""
+    name = config.get(CONF_NAME)
+    host = config.get(CONF_HOST)
+
+    add_devices([PhilipsJS(host, name)])
+
+
+# pylint: disable=too-many-instance-attributes,abstract-method
```

Comment:
`too-many-instance-attributes` is now [globally](https://github.com/home-assistant/home-assistant/blob/dev/pylintrc) disabled.